src/hooks/useAnonymousLogin.ts

```diff
@@ -6,13 +6,21 @@ import { JWTToken } from '../../types/JWT'
 import { useSelector } from 'react-redux'
 import { selectToken } from '../components/App/userSelector'
 import { useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import {
+    hideBackdrop,
+    showBackdrop,
+} from '../components/App/Backdrop/backdropActions'
 
 export default async (): Promise<void> => {
     const token = useSelector(selectToken)
+    const dispatch = useDispatch()
```

Comment:
it's weird to have the backdrop behavior for the login splitted in many files, (login and token). 
Here a single action could do everything: 

```
const signInAnonymouslyWithToken = () => async (dispatch, getState) => {
if(isUserSignIn(getState())) {
    return
}

                dispatch(showBackdrop())

                try {
                    if (isAuthEmulatorEnabled()) {
                        authInstance.useEmulator(EMULATORS.hosts.auth)
                    }
                    await firebase.auth().signInAnonymously()
                    console.log('User signed up anonymously...')
                    
                    await dispatch(getToken())
                    
                    dispatch(hideBackdrop())
                } catch (error) {
                    console.log(
                        `Error anonymous login: ${error.message}, code: ${error.code}`
                    )
                    dispatch(hideBackdrop())
                }

}

```

Reply:
why splitted ? You have both `show` and `hide` within `useAnonymousLogin.ts`. I think the one that is in `Login` concerns another "loading"

Reply:
I said splitted because the backdrop is not hidhen on success there and this is managed somewhere else; which is not obvious.